feat(ui-kit) : migrating userProfile components , their storybook and…

ObitoDevv opened the pull request:
fix #32

ObitoDevv commented:
@maany sir, thanks a lot for your support.  Translation is working fine with ui-kit & platform apps , all our components , storybook and testing we have completed. Just one storybook files where the translations are used is creating some error. Can you also run at your end once? I am attaching screenshot of what error is coming. 

![image](https://github.com/user-attachments/assets/b2e58a8e-6705-4636-8389-6aa3056ec951)

Some translation built issue is there its not finding the path.

maany commented:
I just built translations and ui-kit locally. Have you pushed everything? What command will re-create this error?

![image](https://github.com/user-attachments/assets/abb3b7b4-ce8d-444f-8cf5-9f2b0e5547dd)

Also, based on the CI errors, looks like you might have to update the lock file  ( just to a pnpm install to generate it)

ObitoDevv commented:
> I just built translations and ui-kit locally. Have you pushed everything? What command will re-create this error?
> 
> ![image](https://private-user-images.githubusercontent.com/5906242/414314466-abb3b7b4-ce8d-444f-8cf5-9f2b0e5547dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk5NDE2MjIsIm5iZiI6MTczOTk0MTMyMiwicGF0aCI6Ii81OTA2MjQyLzQxNDMxNDQ2Ni1hYmIzYjdiNC1jZThkLTQ0NGYtOGNmNS05ZjJiMGU1NTQ3ZGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMTlUMDUwMjAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDk1ZmMzNTYyZDY0MDk5OWZkZDE0ZmQ2YjI0Y2Y3Y2Y0ZGM5NzBlOWExYjk5Y2EwMGE2N2JlZmQwOWFmNDg1YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aWFV1SG_Zu63cUvrXxOUUst68et9tnSZ8vObRog4hO8)
> 
> Also, based on the CI errors, looks like you might have to update the lock file ( just to a pnpm install to generate it)

ObitoDevv commented:
Issue_1 :
error - in line[5] : 
Cannot find module '@tailwindcss/vite' or its corresponding type declarations.
  There are types at 'g:/risee11/swiss/siwss-2/dad-e-class-monorepo-3/packages/ui-kit/node_modules/@tailwindcss/vite/dist/index.d.mts', but this result could not be resolved under your current 'moduleResolution' setting. Consider updating to 'node16', 'nodenext', or 'bundler'.ts(2307)
  
  Solution - Option 'module' must be set to 'Node16' when option 'moduleResolution' is set to 'Node16'.

![image](https://github.com/user-attachments/assets/25bbc18e-cc21-46b2-8d44-b330557a17f2)


Issue_2: We are not able to import the models and translations in the stories and tests folder - files (ui-kit). 

![image](https://github.com/user-attachments/assets/1f6a4676-7653-451b-b998-1b907fbfa5e3)

Issue_3 : We are not able to use the alias path in the .storybook/main.ts (in ui-kit).

![image](https://github.com/user-attachments/assets/5f07c156-9e3b-48f1-b44e-8df718e4c91f)

maany commented:
@ObitoDevv I believe I have addressed all the issues we discussed in the call today, CI is passing ;) Can you pull my changes and verify locally that work for you.

maany commented:
@ObitoDevv I have updated the description of this PR, could you complete the checklist :)
Also, should #32 be closed in favor of this PR?

ObitoDevv commented:
@maany Thank you so much for changes. Sir. 
One last issue is coming like in my windows only when I am using -[pnpm exec nx run-many -t lint] - then don't know these error are coming. Can u please help me to resolve this issue?


![image](https://github.com/user-attachments/assets/8266e925-7cfc-41fd-89a1-8e30f110f0c0)

maany commented:
these look to be errors from compiled javascript files in the dist directories or node_modules. It seems like path matching on windows does not work. I would not be able to replicate this easily ( do not have any windows machines).

You could try using a VSCode dev conttainer ( linux based) on your windows machine:

https://code.visualstudio.com/docs/devcontainers/containers

alebg commented:
@ObitoDevv please update this branch with the new changes just merged into main (from #42 )

alebg commented on packages/models/src/profile.ts:
```diff
@@ -58,7 +58,8 @@ export const PersonalProfileSchema = z.discriminatedUnion("isRepresentingCompany
  */
-export type TPersonalProfile = z.infer<typeof PersonalProfileSchema>;
+
+export type TPersonalProfile = z.infer<typeof PersonalProfileSchema> ;
```
@ObitoDevv why is there a space here?

ObitoDevv replied:
Ohh, my bad sir! By mistake one space has added there. Should I remove it ?

alebg replied:
@ObitoDevv No worries! But yes please remove it, otherwise it adds to the amount of files that we need to review

alebg reviewed (commented):
@ObitoDevv thanks!

Alright, so, this PR is supposed to be for userprofile components. Just to undestand, could you please briefly explain why do we have changes to these files?

- apps/platform/tsconfig.json
- packages/models/tsconfig.lib.json
- packages/translations/package.json

was something missing in the translations PR?

ObitoDevv commented:
> @ObitoDevv thanks!
> 
> Alright, so, this PR is supposed to be for userprofile components. Just to undestand, could you please briefly explain why do we have changes to these files?
> 
> * apps/platform/tsconfig.json
> * packages/models/tsconfig.lib.json
> * packages/translations/package.json
> 
> was something missing in the translations PR?

Yes sir, we added references path of models & storybook path in these files. To use the alias to import the models.

alebg commented:
Alright then, if everything's good, @carlapbg please re-review this PR checking that the comments from your first review (which you can find in [the original PR](https://github.com/dream-aim-deliver/dad-e-class-monorepo/pull/32)) were integrated

alebg commented on packages/translations/src/lib/dictionaries/de.ts:
```diff
@@ -1,10 +1,88 @@
+      date: 'Datum',
+      checkboxtext1: 'Ich handle im Namen eines Unternehmens',
+      companyName: 'Firmenname',
+      companyNamePh: 'e.g. Acme',
```
@ObitoDevv for consistency with your naming scheme, please expand these "Ph" to "Placeholder" (for `companyNamePh`, `companyUIDPh`, and any other)

alebg commented on packages/ui-kit/lib/components/profile/professional-info.tsx:
```diff
@@ -0,0 +1,360 @@
+    () => formData.skills ?? [],
+  );
+
+  // const [allSkills, setAllSkills] = React.useState<string[]>([]);
```
@ObitoDevv please don't leave dead code like this, if unused just delete it

alebg reviewed (changes requested):
@ObitoDevv A couple comments:

1. I still get the gray calendar icon from Firefox on the date input field:
![immagine](https://github.com/user-attachments/assets/5677c37a-e2d6-4cbe-97b3-d773c69de2e0)
In the original PR there was an InputField story that also showed this for the date, and there was an additional DateInput story too.

 Apart from this, and my in-line comments, I don't have any other observation. Thanks a lot!

ObitoDevv commented:
> @ObitoDevv A couple comments:
> 
> 1. I still get the gray calendar icon from Firefox on the date input field:
>    ![immagine](https://private-user-images.githubusercontent.com/62880920/416398117-5677c37a-e2d6-4cbe-97b3-d773c69de2e0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA0NDY4NTYsIm5iZiI6MTc0MDQ0NjU1NiwicGF0aCI6Ii82Mjg4MDkyMC80MTYzOTgxMTctNTY3N2MzN2EtZTJkNi00Y2JlLTk3YjMtZDc3M2M2OWRlMmUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjI1VDAxMjIzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlYmIzYjJkZTc3Zjk1OWU4YmVhM2ZiZmNmNWE4NTEyNzRmMGI5YTI4NGE2NDQ4ZGNjNmEwZDRkOGE4MTEwMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XUQLVVxUNP_mEAjhQtBSEyKGmbXyOfEgzL--xqA1u7Y)
>    In the original PR there was an InputField story that also showed this for the date, and there was an additional DateInput story too.
> 
> Apart from this, and my in-line comments, I don't have any other observation. Thanks a lot!

@alebg  Sir, thank you for the feedback. I've addressed the issue with the Firefox date picker icon. Previously, we were using CSS pseudo-element selectors to hide the default calendar icon, which worked in other browsers but not in Firefox.
The problem was that Firefox's native calendar icon (::moz-calendar-picker-indicator) wasn't being properly hidden with our original CSS approach.
I've implemented a more robust solution that:

Makes the native input element invisible while preserving its functionality
Creates a custom display layer that shows the formatted date
Ensures clicking anywhere in the field triggers the date picker
Positions the calendar dropdown correctly below the input field

The calendar icon should now be completely hidden in Firefox while maintaining the custom UI appearance and functionality across all browsers. The component still uses the native date picker implementation for best accessibility and user experience.
Let me know if you notice any other browser-specific issues!

carlapbg commented on packages/ui-kit/lib/components/button.tsx:
The buttons must have a fixed height in rem, and padding should only be added horizontally.

carlapbg commented on packages/ui-kit/lib/components/checkbox.tsx:
- The text of the checkbox should be in “color-text-primary”: #FAFAF9
- When I use the Safari browser, I can't see the black arrow when I click on the checkbox.

carlapbg commented on packages/ui-kit/lib/components/radio-button.tsx:
The text of the checkbox should be in “color-text-primary”: #FAFAF9

ObitoDevv replied:
Same as we did for checkbox. We did color-text-primary here also.

carlapbg commented on packages/ui-kit/lib/components/tabs/tab-list.tsx:
In the Storybook, only one variant appears. In Figma, there are the small and default variants.

<img width="825" alt="Screenshot 2025-02-25 at 12 27 51" src="https://github.com/user-attachments/assets/0c63506d-8402-4f9c-af9b-5e38516e6291" />

carlapbg commented on packages/ui-kit/lib/components/drag&drop/uploaded-file.tsx:
The gap between the file name and the cloud-download and trash icons should have more distance.

This is how it should look:
<img width="663" alt="Screenshot 2025-02-25 at 12 31 49" src="https://github.com/user-attachments/assets/d60fb2f3-72ba-4d15-8a85-feba14203a9b" />

carlapbg commented on packages/ui-kit/lib/components/drag&drop/uploaded-file.tsx:
When there is a file with a very long name, it should display the full name on hover. The current setting doesn't limit how the name is shown. If the name is too long, the best approach would be to show the full name only on hover.

carlapbg commented on packages/ui-kit/lib/components/drag&drop/uploaded-image.tsx:
- Same as uploaded-file.tsx: There should be more distance between the file and the icons. They should be aligned to the right side
- When there is a file with a very long name, it should display the full name on hover.

carlapbg reviewed (changes requested):
Hi guys, good job, thanks! I was reviewing the components and I have left some comments.

maany commented:
force-pushed after resolving merge conflicts *
